content/en/docs/How Tos/metrics.md

```diff
@@ -134,13 +73,160 @@ or an error saying why it can't.
+    - `NumProcessedFiles`, the total number of processed files since the component creation
+    - `NumErrorFiles`, the number of files that have produced an error
+
+Due to historical reasons, these fields have the word _lines_ in them but they do 
+mean the number of records.
+
+The code example of 
+[how-to-create-filter](https://getbaker.io/docs/how-tos/create_filter/#processing-records) 
+shows how to correctly reports metrics in a custom Filter.
+
+### Report custom metrics
+
+Components may need to report custom metrics for monitoring some specific events. 
+Baker supports two main ways to expose custom metrics, namely:
+
+- return a 
+[`baker.MetricsBag`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsBag)
+instance from the `Stats` method 
+- directly use the 
+[`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient) in 
+the component code
+
+The two mechanisms follow the pull vs push approach respectively.
+The `MetricsBag` should be returned by the `Stats` method along with the default
+statistics and it will be collected once per second by the Baker runtime. Differently,
+the `MetricsClient` can be requested from the Baker topology in the component
+instantiation and it can be used in any part of the component code to report metrics.
+If there is no particular requirement it is suggested to prefer the `MetricsBag` approach. 
+Indeed, the pull approach permits the reduction of the metrics overhead during the 
+record processing.
+
+Both `MetricsBag` and `MetricsClient` Metrics support the most common metric types,
+namely:
+- `RawCounter`: a cumulative counter that can only increase
+- `DeltaCounter`: the total number of event occurrences in a unit time
+- `Gauge`: a snapshot of an event in the last unit time
+- `Histogram`: a statistical distribution of a set of values in one unit of time
+- `Duration` or `Timing`: like an histogram but with time durations
+
+Moreover, the `MetricsClient` is the direct interface to the remote metrics service 
+used by the topology, e.g., *Datadog*, thus it supports some specific features such 
+as **tags**.
+Tags are a way of adding dimensions to telemetries so they can be filtered, 
+aggregated, and compared in different visualizations.
+Therefore, if a component requires to publish its metrics with a set of specific tags, the 
+`MetricsClient` should be used rather than `MetricsBag`.
+However, it is suggested to use the `MetricsClient` in the `Stats` method to limit
+the number of calls to the client. Indeed, calling the `MetricsClient` methods inside the 
+`Process` method could introduce overheads and performance degradations.
+
+In summary, the go-to way is to implements custom statistics with a `MetricsBag`, but 
+there are some situations in which it is preferable the `MetricsClient`, namely:
+- the components need to publish the metrics using specific tags,
+- the components cannot centralize the metric collections, e.g. the component has 
```

Comment:
```suggestion
- you can't centralize metrics collection in the `Stats` method since the metrics to expose are produced by different worker goroutines running inside your component 
```